src/hotspot/share/ci/ciField.cpp

```diff
@@ -253,6 +253,22 @@ static bool trust_final_non_static_fields(ciInstanceKlass* holder) {
   return TrustFinalNonStaticFields;
 }
 
+// Todo: Change to 'java/lang/StableValue' etc. once StableValue becomes a public API
+const char* stable_value_klass_name = "jdk/internal/lang/StableValue";
+const char* stable_array_klass_name = "jdk/internal/lang/StableArray";
+const char* stable_array2d_klass_name = "jdk/internal/lang/StableArray2D";
+const char* stable_array3d_klass_name = "jdk/internal/lang/StableArray3D";
+
+static bool trust_final_non_static_fields_of_type(Symbol* signature) {
```

Comment:
Is there a better way of doing this?

Reply:
How do we check if the type implements `TrustedFieldType` in C?

Reply:
Is it possible for us to just look at strict fields from valhalla, so we can reliably constant-fold those strict final fields? https://cr.openjdk.org/~jrose/jls/constructive-classes.html